packages/peer-exchange/src/waku_peer_exchange_discovery.ts

```diff
@@ -109,6 +129,35 @@ export class PeerExchangeDiscovery
     return "@waku/peer-exchange";
   }
 
+  private readonly startRecurringQueries = async (
+    peerId: PeerId,
+    firstQueryTimeout: number = DEFAULT_PEER_EXCHANGE_INITIAL_QUERY_TIMEOUT_MS,
+    attemptsBeforeBackoff: number = DEFAULT_ATTEMPTS_BEFORE_BACKOFF,
+    queryInterval: number = DEFAULT_PEER_EXCHANGE_QUERY_INTERVAL_MS
+  ): Promise<void> => {
+    const peerIdStr = peerId.toString();
+    if (!this.queryAttempts.has(peerIdStr))
+      this.queryAttempts.set(peerIdStr, 1);
+
+    setTimeout(async () => {
+      await this.query(peerId);
+    }, firstQueryTimeout);
```

Comment:
```suggestion
    setTimeout(this.query(peerId), firstQueryTimeout);
```
Why the wrapping?

Reply:
I still don't understand the point of the wrapper. Why not doing the code change I proposed? or is it a return type issue because `setTimeout` expects the first argument to return `Promise<void>`?